eng/Directory.Build.Data.targets

```diff
@@ -23,24 +23,6 @@
     </PropertyGroup>
   </Target>
 
-  <!-- Validates that there is a release note entry for the current package version -->
```

Comment:
One side-effect of removing this logic is that local devs will not get any feedback that they are missing this until they submit a PR. I actually think once you address my other feedback in your tools PR we should simply change this to call the verify-changelog script here but still keep it here to allow devs to get early feedback.